src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingletonProxy.cs

```diff
@@ -126,19 +167,40 @@ public ClusterSingletonProxy(string singletonManagerPath, ClusterSingletonProxyS
                         _singleton = subject;
                         Context.Watch(subject);
                         CancelTimer();
+                        Context.System.EventStream.Publish(IdentifySingletonResult.Success(
+                            singletonName: _settings.SingletonName, 
+                            role: _settings.Role));
                         SendBuffered();
                     }
                 });
             Receive<TryToIdentifySingleton>(_ =>
                  {
                      var oldest = _membersByAge.FirstOrDefault();
-                     if (oldest != null && _identityTimer != null)
+                     if (oldest != null && _isIdentifying)
                      {
                          var singletonAddress = new RootActorPath(oldest.Address) / _singletonPath;
                          Log.Debug("Trying to identify singleton at [{0}]", singletonAddress);
                          Context.ActorSelection(singletonAddress).Tell(new Identify(_identityId));
                      }
                  });
+            Receive<IdentifySingletonTimeOutTick>(_ =>
+            {
+                // We somehow missed a CancelTimer() and a singleton reference was found when we waited,
+                // ignoring the timeout tick message.
+                if (_singleton is not null)
+                {
+                    Timers.Cancel(IdentifySingletonTimeOutTick.Instance);
+                    return;
+                }
+                
+                Log.Warning(
+                    "ClusterSingletonProxy failed to find an associated singleton named [{0}] in role [{1}] after {2} seconds.",
+                    _settings.SingletonName, _settings.Role, _settings.SingletonIdentificationFailurePeriod.TotalSeconds);
+                
+                Context.System.EventStream.Publish(IdentifySingletonResult.Timeout(
```

Comment:
Yes, I think it is a good addition, people can use this to detect if something went wrong in their application by hooking to this event, instead of having to check their logs